src/main/webapp/app/exercises/shared/plagiarism/plagiarism-inspector/plagiarism-inspector.component.ts

```diff
@@ -13,35 +13,97 @@ import { ExportToCsv } from 'export-to-csv';
+     */
     modelingExercise: ModelingExercise;
+
+    /**
+     * Results of the plagiarism detection.
+     */
     modelingSubmissionComparisons: Array<ModelingSubmissionComparisonDTO>;
-    splitControlSubject: Subject<string> = new Subject<string>();
 
-    private subscription: Subscription;
+    /**
+     * Flag to indicate whether the plagiarism detection is currently in progress.
+     */
+    plagiarismDetectionInProgress: boolean;
+
+    /**
+     * Index of the currently selected plagiarism.
+     */
+    selectedPlagiarismIndex: number;
+
+    /**
+     * Subject to be passed into PlagiarismSplitViewComponent to control the split view.
+     */
+    splitControlSubject: Subject<string> = new Subject<string>();
 
     constructor(private route: ActivatedRoute, private router: Router, private modelingExerciseService: ModelingExerciseService) {}
 
     ngOnInit() {
-        this.subscription = this.route.params.subscribe((params) => {
-            this.modelingExerciseService.find(params['exerciseId']).subscribe((response: HttpResponse<ModelingExercise>) => {
-                this.modelingExercise = response.body!;
-            });
+        this.route.params.subscribe((params) => {
+            this.fetchModelingExercise(params['exerciseId']);
+        });
+    }
+
+    /**
+     * Fetch the modeling exercise with the given id.
+     *
+     * @param modelingExerciseId
+     */
+    fetchModelingExercise(modelingExerciseId: number) {
+        this.modelingExerciseService.find(modelingExerciseId).subscribe((response: HttpResponse<ModelingExercise>) => {
+            this.modelingExercise = response.body!;
         });
     }
 
-    handleTagPlagiarism(confirmed: boolean) {
-        this.modelingSubmissionComparisons[this.selectedComparisonIndex].confirmed = confirmed;
+    /**
+     * Handle the 'plagiarismStatusChange' event emitted by PlagiarismHeaderComponent.
+     *
+     * @param confirmed
+     */
+    handlePlagiarismStatusChange(confirmed: boolean) {
+        this.modelingSubmissionComparisons[this.selectedPlagiarismIndex].confirmed = confirmed;
+    }
+
+    /**
+     * Handle the 'splitViewChange' event emitted by PlagiarismHeaderComponent.
+     *
+     * @param pane
+     */
+    handleSplitViewChange(pane: string) {
+        this.splitControlSubject.next(pane);
+    }
+
+    /**
+     * Trigger the server-side plagiarism detection and fetch its result.
+     */
+    checkPlagiarism() {
+        this.plagiarismDetectionInProgress = true;
+
+        this.modelingExerciseService.checkPlagiarism(this.modelingExercise.id!).subscribe(
+            (response: HttpResponse<Array<ModelingSubmissionComparisonDTO>>) => {
+                this.plagiarismDetectionInProgress = false;
+                this.modelingSubmissionComparisons = response.body!.sort((c1, c2) => c2.similarity - c1.similarity);
+            },
+            () => (this.plagiarismDetectionInProgress = false),
+        );
```

Comment:
We usually have networking code in a service class. I'd argue that would also improve the structure here.

Reply:
Mapped the `response` to its `body` attribute in `modelingExerciseService.checkPlagiarism(...)` now, so I can work with the comparison results in the component directly.